docs/configuration.md

```diff
@@ -0,0 +1,214 @@
+
+```
+export CHECK_CORRUPTION_ON_APPEND=1
+```
+
+
+
+<br><br>
+
+
+## Serialization
+
+### ARCTIC_AUTO_EXPAND_CHUNK_SIZE
+
+If a row is too large, then auto-expand the data chunk size from the default _CHUNK_SIZE (it is 2MB). It is disabled by default, and the written DataFrame in its serialized Numpy array form shouldn't exceed 2MB.
+
+This setting is effective only when using the incremental serializer.
+
+```
+export ARCTIC_AUTO_EXPAND_CHUNK_SIZE=1
+```
+
+
+### MAX_DOCUMENT_SIZE
+
+This configuration variable is used only when ARCTIC_AUTO_EXPAND_CHUNK_SIZE is set and the user writes a DataFrame which has extremely large number of columns (exceed 2MB serialized).
+
+This value must be less than 16MB which is the maximum document size of MongoDB, taking into account the size of other fields, serialized as a BSON object.
+
+Its default value is configured as follows:
+
+```
+In[8]: pymongo.common.MAX_BSON_SIZE * 0.8
+Out[8]: 13421772.8
+```
+
+
+### FAST_CHECK_DF_SERIALIZABLE
+
+Optional optimisation feature. When set, it applies a fast check for the *can_write()* of the Pandas-specific store implementations. This check takes place to decide the right write handler, among the registered ones, applied on the data provided by the user to be written/appended. It has a significant impact for large DataFrames, which have columns with object dtype. The benefits are even more evident if the number of object columns is proportionally small to the total number of columns.
+
+```
+export FAST_CHECK_DF_SERIALIZABLE=1
+```
+
+
+
+
+
+<br><br>
+
+
+
+## Forward pointers
+
+### ARCTIC_FORWARD_POINTERS_CFG
+
+This feature flag controls the mode of operation for the data segment references model.
+
+The original implementation of Arctic stores the *_id* values of version document inside the data segment documents (parent references). Therefore, even small updates (i.e. writes where data change very little) cause a large number of documents to be updated, and trigger fetches in the WiredTiger cache (e.g. when appending).
+
+Since Arctic version *1.73.0* the segment referencing model of VersionStore has included a new implementation, named as *forward pointers*. In this model, the segments no longer hold information about the versions which reference/use them, but instead, the list of segment SHAs is stored in the version document itself. This is beneficial in many ways. First, small updates resultin updates/writes only for the new data segments, reducing dramatically the number of affected documents by the update queries (faster execution). Second, the WiredTiger cache gets less polluted utilizing the necessary indexes, not fetching existing large data segments in cache. Finally, all the data information is now in one place, the version document itself, making the debugging of data integrity issues easier.
```

Comment:
```suggestion
Since Arctic version *1.73.0* the segment referencing model of VersionStore has included a new implementation, named as *forward pointers*. In this model, the segments no longer hold information about the versions which reference/use them, but instead, the list of segment SHAs is stored in the version document itself. This is beneficial in many ways. First, small updates result in updates/writes only for the new data segments, reducing dramatically the number of affected documents by the update queries (faster execution). Second, the WiredTiger cache gets less polluted utilizing the necessary indexes, not fetching existing large data segments in cache. Finally, all the data information is now in one place, the version document itself, making the debugging of data integrity issues easier.
```

Reply:
thanks